src/circuit_breaker.erl

```diff
@@ -252,6 +255,14 @@ do_call(Service, CallFun, CallTimeout, ResetFun, ResetTimeout, Thresholds) ->
                     ResetTimeout, Thresholds)
   end.
 
+preserve_exception(CallFun) ->
+  try
+    CallFun()
+  catch
+    Class:Reason ->
+      exit({raise, Class, Reason, erlang:get_stacktrace()})
```

Comment:
it's deprecated in otp 21. rather avoid using it at all ?

Reply:
I'd rather keep the possibility of preserving the stacktrace.

I could write two versions of this, using the new `OTP_RELEASE` macro that is introduced in OTP 21:
```
-ifdef(OTP_RELEASE).
%% version with Class:Reason:Stacktrace
-else.
%% current version
-endif.
```
Would that be good enough?

Reply:
yes, that'll work.

Reply:
Done